airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation/documentation.py

```diff
@@ -0,0 +1,605 @@
+                message=textwrap.dedent(
+                    f"""
+                Migration guide file for {connector.name_from_metadata} has incorrect version headings.
+                Check for missing, extra, or misordered headings, or headers with typos.
+                Expected headings: {ordered_expected_headings}
+                """
+                ),
+            )
+        return self.create_check_result(
+            connector=connector,
+            passed=True,
+            message="The migration guide is correctly templated",
+        )
+
+
+class CheckDocumentationExists(DocumentationCheck):
+    name = "Connectors must have user facing documentation"
+    description = (
+        "The user facing connector documentation should be stored under `./docs/integrations/<connector-type>s/<connector-name>.md`."
+    )
+
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="User facing documentation file is missing. Please create it under ./docs/integrations/<connector-type>s/<connector-name>.md",
+            )
+        return self.pass_(
+            connector=connector,
+            message=f"User facing documentation file {connector.documentation_file_path} exists",
+        )
+
+
+class CheckDocumentationContent(DocumentationCheck):
+    """
+    For now, we check documentation structure for sources with sl >= 300.
+    """
+
+    applies_to_connector_languages = [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]
+    applies_to_connector_ab_internal_sl = 300
+    applies_to_connector_types = ["source"]
+
+
+class CheckDocumentationLinks(CheckDocumentationContent):
+    name = "Links used in connector documentation are valid"
+    description = f"The user facing connector documentation should update invalid links in connector documentation."
+    valid_status_codes = [200, 403, 401, 405, 429, 503]  # we skip 4xx due to needed access
+
+    def validate_links(self, connector: Connector) -> List[str]:
+        errors = []
+        threads = []
+
+        def request_link(docs_link: str) -> None:
+            try:
+                response = requests.get(docs_link)
+                if response.status_code not in self.valid_status_codes:
+                    errors.append(f"{docs_link} with {response.status_code} status code")
+            except requests.exceptions.SSLError:
+                pass
+            except requests.exceptions.ConnectionError:
+                pass
+
+        for link in DocumentationContent(connector=connector).links:
+            process = Thread(target=request_link, args=[link])
+            process.start()
+            threads.append(process)
+
+        for process in threads:
+            process.join(timeout=30)  # 30s timeout for process else link will be skipped
+            process.is_alive()
+
```

Comment:
Isn't the use of `Thread` overkill here? Why not use `request.get(docs_link, timeout=30)`?

Reply:
for some documentations that have a lot of links, for example in changelog section, it makes this test work faster. seconds instead of minutes.